tests/conftest.py

```diff
@@ -473,7 +473,11 @@ def __call__(self, dag_id='test_dag', session=None, **kwargs):
             self.kwargs = kwargs
             self.session = session
             self.start_date = self.kwargs.get('start_date', None)
+            default_args = kwargs.get('default_args', None)
+            if not self.start_date and 'start_date' in default_args:
+                self.start_date = default_args.get('start_date')
```

Comment:
Gosh this `default_args` thing is really awkward 🤯

Reply:
Would it work if we do this?

```python
def __call__(self, dag_id='test_dag', default_args=None, session=None, **kwargs):
    if default_args is not None:
        kwargs = {**default_args, **kwargs}
    self.kwargs = kwargs
    self.session = session
    self.start_date = self.kwargs.get('start_date', None)
```

Reply:
The default_args here is the one used by DAG to pass to tasks. It can be in the kwargs as `default_args`.
See: https://github.com/apache/airflow/blob/37cc115d30dc17cf474a6eff04aad056cb6adef4/airflow/models/dag.py#L165